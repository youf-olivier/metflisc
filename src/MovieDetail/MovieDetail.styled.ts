import { device } from 'shared/utilities/devise.constants';
import styled from 'styled-components';

export const StyledImage = styled.img`
  width: 48%;
  margin: 0.1em;
  @media ${device.tablet} {
    width: 24%;
  }

  @media ${device.laptop} {
    width: 18%;
  }
`;

export const StyledArticle = styled.article`
  display: flex;
  flex-wrap: wrap;
  padding: 1em 0.2em;
  flex-direction: column;
  align-items: center;
  font-weight: 500;
  color: ${props => props.theme.colors.font};
  @media ${device.tablet} {
    align-items: flex-start;
    flex-direction: row;
    max-width: 950px;
    margin: auto;
  }
`;

export const StyledSection = styled.section`
  margin: 1em;
  @media ${device.tablet} {
    width: 70%;
  }
`;
