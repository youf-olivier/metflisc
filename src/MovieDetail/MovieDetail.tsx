import { FC } from 'react';

type MovieDetailProps = {
  title: string;
  overview: string;
  popularity: number;
  posterPath: string;
  releaseDate: Date;
};

const MovieDetail: FC<MovieDetailProps> = ({
  title,
  overview,
  popularity,
  posterPath,
  releaseDate,
}) => (
  <>
    <span>Page DÃ©tail</span>
    <span>{title}</span>
    <span>{overview}</span>
    <span>{popularity}</span>
    <span>{posterPath}</span>
    <span>{releaseDate}</span>
  </>
);

export default MovieDetail;
