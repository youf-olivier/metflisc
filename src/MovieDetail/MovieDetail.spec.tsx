import { render } from '@testing-library/react';

import MovieDetail from './MovieDetail';

describe('MovieDetail tests suite', () => {
  it('should render without crashing', () => {
    const { asFragment } = render(<MovieDetail />);
    expect(asFragment()).toMatchSnapshot();
  });
});
