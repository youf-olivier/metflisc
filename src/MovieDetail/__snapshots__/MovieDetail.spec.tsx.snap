// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetail tests suite should render without crashing 1`] = `
<DocumentFragment>
  <span>
    Page DÃ©tail
  </span>
  <span />
  <span />
  <span />
  <span />
  <span />
</DocumentFragment>
`;
